examples/va/nvidia-mdev/dataplane.md

```diff
@@ -0,0 +1,80 @@
+# Configuring and deploying the dataplane
+
+## Assumptions
+
+- The [control plane](control-plane.md) has been created and successfully deployed
+
+## Initialize
+
+Switch to the "openstack" namespace
+```
+oc project openstack
+```
+Change to the nvidia-mdev/edpm directory
+```
+cd architecture/examples/va/nvidia-mdev/edpm
+```
+Edit the [nodeset/values.yaml](nodeset/values.yaml) and [deployment/values.yaml](deployment/values.yaml) files to suit 
+your environment.
+```
+vi nodeset/values.yaml
+vi deployment/values.yaml
+```
+Generate the dataplane nodeset CR.
+```
+kustomize build nodeset > dataplane-nodeset.yaml
+```
+Generate the dataplane deployment CR.
+```
+kustomize build deployment > dataplane-deployment.yaml
+```
+
+## Create CRs
+Create the nodeset CR
+```
+oc apply -f dataplane-nodeset.yaml
+```
+Wait for dataplane nodeset setup to finish
+```
+oc wait osdpns openstack-edpm --for condition=SetupReady --timeout=600s
+```
+
+Start the deployment
+```
+oc apply -f dataplane-deployment.yaml
+```
+
+Wait for dataplane deployment to finish
+```
+oc wait osdpns openstack-edpm --for condition=Ready --timeout=60m
+```
+
+## Apply necessary Nvidia Configurations
+Update or create a blacklist file in /etc/modeprobe
+```
+cat /etc/modprobe.d/blacklist-nouveau.conf
+blacklist nouveau
+options nouveau modeset=0
+```
+Download the relevant Nvidia driver for your hardware and install per Nvidia's
+instructions.
+Regenerate initramfs
+```
+dracut --force
+grub2-mkconfig -o /boot/grub2/grub.cfg --update-bls-cmdline
+```
```

Comment:
I'm fine to merge this PR with the manual steps above since it's useful to know how to deploy with the nvidia.

In the long run maybe this VA could include installing an EDPM service via the kustomize output to do these steps for the user. We shouldn't block on that though.

https://openstack-k8s-operators.github.io/openstack-operator/dataplane/#_composable_services

Reply:
I agree, I think migrating what is done in cifmw over here in a composable service would be a good next step.